platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java

```diff
@@ -433,6 +433,12 @@ public static void checkNodesToRun(@NonNull final Collection<NodeId> nodesToRun)
                     new NodeAddressMismatchPayload(Network.getInternalIPAddress(), externalIpAddress));
             exitSystem(NODE_ADDRESS_MISMATCH);
         }
+        if (!allowMultipleNodes && nodesToRun.size() > 1) {
+            logger.error(
+                    EXCEPTION.getMarker(),
+                    "Multiple nodes are configured to run. Only one node can be started per java process.");
+            exitSystem(NODE_ADDRESS_MISMATCH);
+        }
```

Comment:
This check needs to move up to the `ServicesMain` class after we determine if there is a node ID passed in the command line args.